src/Set.php

```diff
@@ -24,6 +24,7 @@ final class Set
      */
     private function __construct(
         private Implementation $implementation,
+        private bool $unbounded,
```

Comment:
Since the `Bounded` implementation knows if the value went through a bounding implementation, tracking this here is not necessary. The only advantage is to possibly remove a layer from the composition stack.

But this `Bounded` layer is here only for backward compatibility and will be remove in the future major version. 

Conclusion: this flag is pretty much useless and should be removed.